src/System.Text.Json/src/System/Text/Json/Serialization/WriteStackFrame.cs

```diff
@@ -46,29 +46,14 @@ public void Initialize(Type type, JsonSerializerOptions options)
 
         public void WriteObjectOrArrayStart(ClassType classType, Utf8JsonWriter writer, bool writeNull = false)
         {
-            if (JsonPropertyInfo?.EscapedName != null)
+            if (JsonPropertyInfo?.EscapedName.HasValue == true)
```

Comment:
nit: if we are keeping nullable, this condition can be simplified.

```suggestion
            if (JsonPropertyInfo?.EscapedName.HasValue)
```

Reply:
@ahsonkhan I have tried that, the compiler does not handle it well, it fails with the following error if I apply your suggestion:

```log
System\Text\Json\Serialization\WriteStackFrame.cs(49,17): error CS0266: Cannot implicitly convert type 'bool?' to 'bool'. An explicit conversion exists (are you missing a cast?)
```

Reply:
That's strange. `HasValue` returns a `bool`, not `bool?`

Reply:
Hmm so `if (JsonPropertyInfo.EscapedName.HasValue)` works, but not `if (JsonPropertyInfo?.EscapedName.HasValue)`

Reply:
@ahsonkhan You can give it a try with [sharplab.io](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+ABAzAAmwCZ8BhAWACgBvK/e/ASwDsYB+fZgVwBseJgPOAG46DMfTyEALPgCyAClL4A9jAAWiAJQT8tSg0NMAZvgVrNCNgDpufAUOsAJaADUIPLnB0Gj9fX70AL66IZRBQA=)

```cs
public class C
{
    int? nullable;
    
    public void M(C other)
    {
        if (other?.nullable.HasValue)
        {
        }
    }
}
```

Reply:
Yep, I saw the issue. This goes back to my other feedback, we could just not make it nullable:
https://github.com/dotnet/corefx/pull/37909#discussion_r287544314